operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -385,29 +385,31 @@ var errProgramNotFound = fmt.Errorf("unable to retrieve program for stack")
 //+kubebuilder:rbac:groups=source.toolkit.fluxcd.io,resources=ocirepositories,verbs=get;list;watch
 //+kubebuilder:rbac:groups=auto.pulumi.com,resources=workspaces,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=auto.pulumi.com,resources=updates,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=create;get;list;watch
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;update;patch
 
 // Reconcile reads that state of the cluster for a Stack object and makes changes based on the state read
 // and what is in the Stack.Spec
 func (r *StackReconciler) Reconcile(ctx context.Context, request ctrl.Request) (res ctrl.Result, reterr error) {
 	log := log.FromContext(ctx)
-	log.Info("Reconciling Stack")
 
 	// Fetch the Stack instance
 	instance := &pulumiv1.Stack{}
 	err := r.Client.Get(ctx, request.NamespacedName, instance)
+	if apierrors.IsNotFound(err) {
+		// Request object not found, could have been deleted after reconcile request.
+		// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		// Return and don't requeue
+		log.Info("Stack resource not found. Ignoring since object must be deleted.")
+		return reconcile.Result{}, nil
+	}
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			log.Info("Stack resource not found. Ignoring since object must be deleted.")
-			return reconcile.Result{}, nil
-		}
```

Comment:
Just decreasing indentation.

Reply:
I'm a believer, but looks like the comments are boilerplate so let's do the even more concise thing of
```go
	if err != nil {
		return ctrl.Result{}, client.IgnoreNotFound(err)
	}
```